src/main/java/org/breedinginsight/brapps/importer/services/processors/Processor.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.breedinginsight.brapps.importer.services.processors;
+
+import org.breedinginsight.brapps.importer.model.ImportUpload;
+import org.breedinginsight.brapps.importer.model.imports.BrAPIImport;
+import org.breedinginsight.brapps.importer.model.imports.PendingImport;
+import org.breedinginsight.brapps.importer.model.response.ImportPreviewStatistics;
+import org.breedinginsight.model.Program;
+import org.breedinginsight.services.exceptions.ValidatorException;
+
+import java.util.List;
+import java.util.Map;
+
+public interface Processor {
+
+    /**
+     * Given importRows, get all existing objects in the brapi service. First phase of processing followed by process.
+     * @param importRows
+     * @param program
+     * @throws ValidatorException
+     */
+    //void getExistingBrapiObjects(List<BrAPIImport> importRows, Program program) throws ValidatorException;
+
+    /**
+     * Update mappedBrAPIImport mapping with PendingImport data for brapi object based on new and existing objects.
+     * Return stats on number of new & existing objects
+     * @param importRows
+     * @param mappedBrAPIImport
+     * @param program
+     * @return
+     * @throws ValidatorException
+     */
+    Map<String, ImportPreviewStatistics> process(List<BrAPIImport> importRows,
```

Comment:
I'd suggest adding a `setupProcessor` method.  After looking through some of the processor implementations, there was a pattern of setup that was happening in each.  If there's an explicit method for setup, it will be more readable and maintainable, as well as easier for us to hook in to the different steps for logging and such.

A second suggestion: for each processor implementation, if the methods are made to be package private (meaning you don't specify the package visibility), then only classes in this method can access the processor methods.  This can be helpful to make sure only the `ProcessorManager` is calling processor methods in the correct order.

Reply:
Taking a quick look at package private, looks like that's an issue when implementing interface methods as the interface is public and you cannot reduce visibility in the implementing class.

Reply:
discussed, leaving interface as-is and added back interface method for getting brapi data. Pushed changes